### PR TITLE
Fix healing damage classes and damageable serialization and add test

### PR DESCRIPTION
**Pull request by DrSmugleaf**
Also fixes IDamageableComponent not having a TryGetDamage method for DamageClass, and adds a property for damage properties to support all classes and types automatically.

### CODE REVIEW

File: Content.Shared/Damage/DamageContainer/DamageContainerPrototype.cs [hunk 1]
```diff
@@ -31,18 +32,24 @@ public virtual void LoadFrom(YamlMappingNode mapping)
...
-            var originalTypes = _supportedTypes.ToArray();
+            if (_supportAll)
+            {
+                _supportedClasses.UnionWith(Enum.GetValues<DamageClass>());
+                _supportedTypes.UnionWith(Enum.GetValues<DamageType>());
+                return;
+            }
 
             foreach (var supportedClass in _supportedClasses)
             foreach (var supportedType in supportedClass.ToTypes())
```
Comment (metalgearsloth) [inline Content.Shared/Damage/DamageContainer/DamageContainerPrototype.cs:47]: image

File: Content.Shared/GameObjects/Components/Damage/DamageableComponent.cs [hunk 2]
```diff
@@ -354,7 +354,7 @@ public void Heal()
                     healThisCycle = 0;
 
                     int healPerType;
-                    if (healingLeft > -types.Count)
+                    if (healingLeft < types.Count)
                     {
                         // Say we were to distribute 2 healing between 3
                         // this will distribute 1 to each (and stop after 2 are given)
```
Comment (metalgearsloth) [inline Content.Shared/GameObjects/Components/Damage/DamageableComponent.cs:360]: Github's not letting me comment on it but the line below of healPerType should be 1 given healingLeft is now positive.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Shared/GameObjects/Components/Damage/DamageableComponent.cs (hunk 2 above)
Comment (DrSmugleaf) [reply Content.Shared/GameObjects/Components/Damage/DamageableComponent.cs:360]: Fixed.

File: Content.Shared/Damage/DamageContainer/DamageContainerPrototype.cs (hunk 1 above)
Comment (DrSmugleaf) [reply Content.Shared/Damage/DamageContainer/DamageContainerPrototype.cs]: The nested foreach's have been defeated...

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**